mopidy_tidal/library.py

```diff
@@ -157,104 +151,111 @@ def search(self, query=None, uris=None, exact=False):
+            logger.debug('%r is not available on the backend', uri)
+            return None
+
+        img_uri = self._get_image_uri(item)
+        if not img_uri:
+            logger.debug('%r has no associated images', uri)
+            return None
+
+        logger.debug('Image URL for %r: %r', uri, img_uri)
+        return [Image(uri=img_uri, width=320, height=320)]
+
     def get_images(self, uris):
         logger.info("Searching Tidal for images for %r" % uris)
-        session = self.backend._session
         images = {}
+
         for uri in uris:
             try:
-                uri_images = None
-                if uri.startswith('tidal:'):
-                    parts = uri.split(':')
-                    if parts[1] == 'artist':
-                        artist_id = parts[2]
-                        img_uri = self.lru_artist_img.hit(artist_id)
-                        if img_uri is None:
-                            artist = session.get_artist(artist_id)
-                            img_uri = artist.image
-                            self.lru_artist_img[artist.id] = img_uri
-
-                        uri_images = [Image(uri=img_uri, width=512, height=512)]
-                    elif parts[1] == 'album':
-                        album_id = parts[2]
-                        img_uri = self.lru_album_img.hit(album_id)
-                        if img_uri is None:
-                            album = session.get_album(album_id)
-                            img_uri = album.image
-                            self.lru_album_img[album_id] = img_uri
-
-                        uri_images = [Image(uri=img_uri, width=512, height=512)]
-                    elif parts[1] == 'playlist':
-                        playlist_id = parts[2]
-                        img_uri = self.lru_playlist_img.hit(playlist_id)
-                        if img_uri is None:
-                            playlist = session.get_playlist(playlist_id)
-                            img_uri = playlist.image
-                            self.lru_playlist_img[playlist_id] = img_uri
-
-                        uri_images = [Image(uri=img_uri, width=512, height=512)]
-                    elif parts[1] == 'track':
-                        album_id = parts[3]
-                        img_uri = self.lru_album_img.hit(album_id)
-                        if img_uri is None:
-                            album = session.get_album(album_id)
-                            img_uri = album.image
-                            self.lru_album_img[album_id] = img_uri
-
-                        uri_images = [Image(uri=img_uri, width=512, height=512)]
-                        pass
-
-                images[uri] = uri_images or ()
-            except AttributeError:
-                logger.error("AttributeError when processing URI %r" % uri)
-            except HTTPError as err:
-                logger.error("HTTPError when processing URI %r" % uri)
+                images[uri] = self._get_image(uri)
```

Comment:
Change this line to `images[uri] = self._get_image(uri) or ()` to fix the error `TidalBackend backend returned bad data: Expected a list of Image, not None` when browsing My Playlists, My Albums etc.

Reply:
@tehkillerbee this should be fixed in https://github.com/tehkillerbee/mopidy-tidal/pull/60/commits/b92064caa62d4980185ccb277de31dc6822ba2b6